pyomo/solvers/plugins/solvers/IPOPT.py

```diff
@@ -119,30 +119,32 @@ def create_command_line(self, executable, problem_files):
         # Pyomo/Pyomo) with any user-specified external function
         # libraries
         #
-        if 'PYOMO_AMPLFUNC' in env:
-            if 'AMPLFUNC' in env:
-                env['AMPLFUNC'] += "\n" + env['PYOMO_AMPLFUNC']
+        if "PYOMO_AMPLFUNC" in env:
+            if "AMPLFUNC" in env:
+                for line in env["PYOMO_AMPLFUNC"].split("\n"):
+                    if line not in env["AMPLFUNC"]:
+                        env["AMPLFUNC"] += "\n" + line
```

Comment:
Same comment as above

Reply:
Thanks for looking at this.  I'll make the changes.